cub/benchmarks/bench/select/if.cu

```diff
@@ -155,13 +155,21 @@ void select(nvbench::state& state, nvbench::type_list<T, OffsetT>)
 
   // TODO Extract into helper TU
   const auto selected_elements = thrust::count_if(in.cbegin(), in.cend(), select_op);
-  thrust::device_vector<T> out(selected_elements);
 
   input_it_t d_in                  = thrust::raw_pointer_cast(in.data());
+  output_it_t d_out                = thrust::raw_pointer_cast(in.data());
```

Comment:
important: I'm afraid this can interfere with the benchmark. We have some distribution of input elements that are about to be selected. After the first invocation of the algorithm, all selected items are packed in the beginning of the input. If output actually aliases input, all subsequent invocations will lead to first tiles overwriting their own tiles and subsequent tiles not writing anything back to memory. So memory access pattern changes between iterations of the state.exec. This might increase noise and potentially affect performance results, because most of the tiles will not write anything now. 

Since this is benchmark rather than a test, I believe we can say that we alias data, but provide different iterators. In this case, we'll get a better workload and scope this code to benchmarking overhead of having fences.

Reply:
Sorry, I had also realized this the other day and had already fixed it but just hadn't pushed the changes yet. Yes, totally agree, for the sake of benchmarking we'll just "pretend" it's in-place by passing the `may_alias=true` but keeping input and output buffers separate.